frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js

```diff
@@ -0,0 +1,100 @@
+import { signInAsAdmin, restore, withSampleDataset } from "__support__/cypress";
+
+const QUESTION_NAME = "Cypress Pivot Table";
+
+describe("scenarios > visualizations > pivot tables", () => {
+  beforeEach(() => {
+    restore();
+    signInAsAdmin();
+  });
+
+  it("should be created from an ad-hoc question", () => {
+    // We're intentionally setting the visualization to `bar` here
+    createPivotTableQuestion({ display: "bar" });
+    cy.visit("/collection/root");
+    cy.findByText(QUESTION_NAME).click();
+
+    // By changing the visualization type, we make this an "ad-hoc" question
+    cy.findByText("Visualization").click();
+    cy.get(".Icon-pivot_table")
+      .should("be.visible")
+      .click();
+
+    cy.findByText(/Count by Users? → Source and Products? → Category/); // ad-hoc title
+
+    assertOnPivotSettings();
+    assertOnpivotTable();
+  });
+
+  it("should correctly display saved question", () => {
+    createPivotTableQuestion();
+    cy.visit("/collection/root");
+    cy.findByText(QUESTION_NAME).click();
+
+    assertOnpivotTable();
+
+    // Open Pivot table side-bar
+    cy.findByText("Settings").click();
+
+    assertOnPivotSettings();
+  });
+});
+
+function createPivotTableQuestion({ display = "pivot" } = {}) {
+  withSampleDataset(({ ORDERS, ORDERS_ID, PRODUCTS, PEOPLE }) => {
```

Comment:
This is needed until I merge Cypress metadata extraction updates from the `release-x.37.x` branch.

Reply:
[Done](https://github.com/metabase/metabase/commit/6697758c409d5ec030ad3da456910d49d5381525). Rebased and refactored this function.